Checkout/SPMSample/CheckoutSPMSample.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

```diff
@@ -0,0 +1,16 @@
+{
+  "object": {
+    "pins": [
+      {
+        "package": "CheckoutEventLoggerKit",
+        "repositoryURL": "https://github.com/checkout/checkout-event-logger-ios-framework.git",
+        "state": {
+          "branch": null,
+          "revision": "2ef34662f1525526aeb5681c6af5f69572ab36c9",
+          "version": "1.2.0"
```

Comment:
Disagree here. The Package.swift can include minimal info on versions and allow auto updates if they aren't locked. We should commit this in the same way that we commit our Podfile.lock

Reply:
I would say that this is just a reflection of the latest update from the package dependencies.

If your Package.swift dependency declaration says use latest release, this will update when a new release was made available.

The impact of it will be:
- no effect on restricting the version you use, that comes only from `Package.swift` dependencies declaration
- getting committed constantly without relevance. You may be changing localisations, but the fact that your SPM is resolving the dependencies will update the source controlled file and carry it into your PR if anything was changed, although you haven't technically looked into any differences / impact
- pretty unlikely at this team size, but could lead to merge conflicts. If we have an internal package that has internal dependencies and uses `main` branch (lets say on the SecureDisplay repo I showed yesterday), branches working asynchronously can have different revisions resolved. This can then lead to open PRs that have no overlap but will still create a merge conflict after first PR is merged
- ➕ it does offer at repository level a clear way to view what versions were resolved

This is my interpretation of it, and I would say that it differs from `Podfile.lock` as the pod lock gets updated only when you run a pod install / update, not part of Xcode system build. Packages on the other hand are updated every time you open Xcode and occasionally throughout your use of it. I don't think you can prevent Xcode from resolving Package.swift file as it often times needs to clone dependencies locally and check if they're still in sync (unlike cocoa pods)